bph_pal/bph_pal/__main__.py

```diff
@@ -0,0 +1,63 @@
+# Copyright (c) 2019 Kevin Weiss, for HAW Hamburg  <kevin.weiss@haw-hamburg.de>
+#
+# This file is subject to the terms and conditions of the MIT License. See the
+# file LICENSE in the top level directory for more details.
+# SPDX-License-Identifier:    MIT
+"""packet init for BPH PAL
+This exposes a main function to run the packet from CLI
+"""
+import argparse
+import logging
+
+
+try:
+    from .bph_shell import BphShell as BPH
+except (ImportError, SystemError):
+    try:
+        from bph_pal.bph_shell import BphShell as BPH
+    except (ImportError, SystemError):
+        from bph_shell import BphShell as BPH
+
+
+def main():
+    """
+    Main function to allow direct exec of bph_pal package from CLI
+    """
+    LOG_LEVELS = ('debug', 'info', 'warning', 'error', 'fatal', 'critical')
+    parser = argparse.ArgumentParser(prog='python3 -m bph_pal')
+    parser.add_argument('--loglevel', choices=LOG_LEVELS, default='info',
+                        help='Python logger log level')
+    parser.add_argument('--philip_reset', '-pr', action='store_true',
+                        help='Hard reset of PHiLIP', default=False)
+    parser.add_argument('--load_pinmap', '-lp', default=False,
+                        help='Loads and initializes pinmap')
+    parser.add_argument('--set_pin', '-sp',
+                        help='Sets the pin by key or name')
+    parser.add_argument('--clear_pin', '-clp',
+                        help='Clear the pin by key or name')
+    parser.add_argument('--input_pin', '-inp',
+                        help='Changes pin to input by key or name')
+    parser.add_argument('--power_source', '-ps',
+                        help='Sets the power source mode')
+    pargs = parser.parse_args()
+
+    if pargs.loglevel:
+        loglevel = logging.getLevelName(pargs.loglevel.upper())
+        logging.basicConfig(level=loglevel)
+
+    if pargs.philip_reset:
+        BPH().onecmd("philip_reset")
+    if pargs.load_pinmap:
+        BPH().onecmd("load_pinmap " + str(pargs.load_pinmap))
+    if pargs.set_pin:
+        BPH().onecmd("set_pin " + str(pargs.set_pin))
+    if pargs.clear_pin:
+        BPH().onecmd("clear_pin " + str(pargs.clear_pin))
+    if pargs.input_pin:
+        BPH().onecmd("input_pin " + str(pargs.input_pin))
+    if pargs.power_source:
+        BPH().onecmd("power_source " + str(pargs.power_source))
+
+
+if __name__ == '__main__':
```

Comment:
not needed, because this file is already name `__main__.py` so its implicit that this matches, just call `main()` as in https://github.com/riot-appstore/PHiLIP/blob/master/IF/philip_pal/philip_pal/__main__.py

Reply:
Ahh so you left it out on purpose, good!